File: pku/3627.java
import java.util.*;

public class Main{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int b = sc.nextInt();
        int[] h = new int[n];
        for(int i = 0; i < n;i++)h[i] = sc.nextInt();
        Arrays.sort(h);
        int sum = 0;
        int cur = n-1;
        for(;cur >= 0;cur--){
            sum += h[cur];
            if(sum >= b){
                System.out.println(n-cur);
                break;
            }
        }
    }
}

File: pku/3627.cc
#include <iostream>
#include <algorithm>
using namespace std;

int main(){
  int n,b;
  cin >> n >> b;
  int h[n];
  for(int i = 0; i < n;i++)cin >> h[i];
  int sum = 0;
  sort(h, h+n);
  for(int i = n-1; i>=0;i--){
    sum += h[i];
    if(sum >= b){
      cout << (n-i) << endl;
      break;
    }
  }

  return 0;
}
